File: CPP/Program/ParkingLot.cpp
/*Saya Legi Kuswandi mengerjakan LP4 dalam mata kuliah DPBO
untuk keberkahanNya maka saya tidak melakukan kecurangan seperti yang 
telah dispesifikasikan. Aamiin.*/

#pragma once

#include "Car.cpp"
#include "Motor.cpp"
#include <iostream>
#include <vector>
using namespace std;

class ParkingLot {
private:
    string kapasitas;
    int jumlah_kendaraan_saat_ini;
    vector<Motor> motor;
    vector<Car> mobil;

public:
    // Constructor
    ParkingLot() {}


    ParkingLot(string kapasitas, int jumlah_kendaraan_saat_ini, Motor motor, Car mobil) {
        this->kapasitas = kapasitas;
        this->jumlah_kendaraan_saat_ini = jumlah_kendaraan_saat_ini;
        this->motor.push_back(motor);
        this->mobil.push_back(mobil);
    }

    // Getter and Setter
    string getKapasitas() {
        return kapasitas;
    }

    void setKapasitas(string kapasitas) {
        this->kapasitas = kapasitas;
    }

    int getJumlahKendaraan() { // Getter untuk jumlah kendaraan saat ini
        return jumlah_kendaraan_saat_ini;
    }

    void setJumlahKendaraan(int jumlah) { // Setter untuk jumlah kendaraan saat ini
        this->jumlah_kendaraan_saat_ini = jumlah;
    }

    void addMotor(Motor motor) {
        this->motor.push_back(motor);
        jumlah_kendaraan_saat_ini++; // Tambah jumlah kendaraan saat menambah motor
    }

    vector<Motor> getMotor() {
        return this->motor;
    }

    void addMobil(Car mobil) {
        this->mobil.push_back(mobil);
        jumlah_kendaraan_saat_ini++; // Tambah jumlah kendaraan saat menambah mobil
    }

    vector<Car> getMobil() {
        return this->mobil;
    }

    // Destructor
    ~ParkingLot() {}
};

File: Python/Program/ParkingLot.py
from typing import List
from Car import Car
from Motor import Motor

class ParkingLot:
    def __init__(self, kapasitas="", jumlah_kendaraan_saat_ini=0):
        # Inisialisasi atribut tempat parkir
        self.kapasitas = kapasitas  # Kapasitas tempat parkir
        self.jumlah_kendaraan_saat_ini = jumlah_kendaraan_saat_ini  # Jumlah kendaraan saat ini di tempat parkir
        self.mobil = []  # Daftar mobil yang terparkir
        self.motor = []  # Daftar motor yang terparkir

    # Method untuk menambahkan mobil ke tempat parkir
    def add_mobil(self, mobil):
        self.mobil.append(mobil)
        self.jumlah_kendaraan_saat_ini += 1  # Tambah jumlah kendaraan saat menambah mobil

    # Method untuk menambahkan motor ke tempat parkir
    def add_motor(self, motor):
        self.motor.append(motor)
        self.jumlah_kendaraan_saat_ini += 1  # Tambah jumlah kendaraan saat menambah motor

    # Method untuk mendapatkan kapasitas tempat parkir
    def get_kapasitas(self) -> str:
        return self.kapasitas

    # Method untuk mengatur kapasitas tempat parkir
    def set_kapasitas(self, kapasitas: str):
        self.kapasitas = kapasitas

    # Method untuk mendapatkan jumlah kendaraan saat ini di tempat parkir
    def get_jumlah_kendaraan(self) -> int:
        return self.jumlah_kendaraan_saat_ini

    # Method untuk mengatur jumlah kendaraan saat ini di tempat parkir
    def set_jumlah_kendaraan(self, jumlah: int):
        self.jumlah_kendaraan_saat_ini = jumlah

    # Method untuk mendapatkan daftar motor yang terparkir
    def get_motor(self) -> List[Motor]:
        return self.motor

    # Method untuk mendapatkan daftar mobil yang terparkir
    def get_mobil(self) -> List[Car]:
        return self.mobil
